dashboard/app/models/school.rb

```diff
@@ -420,41 +479,51 @@ def self.merge_from_csv(filename, options = CSV_IMPORT_OPTIONS, update_existing
-
-    if updated_schools_attribute_frequency.any?
-      summary_message <<
-        "Among updated schools, these attributes #{is_dry_run ? 'will be' : 'were'} updated:\n"\
-        "#{updated_schools_attribute_frequency.sort_by {|_, v| v}.
-          reverse.
-          map {|attribute, frequency| attribute + ': ' + frequency.to_s}.join("\n")}\n"
-    end
-
-    # More verbose logging in dry run
-    if is_dry_run
-      if new_schools.any?
+      future_tense_dry_run = is_dry_run ? ' to be' : ''
+      summary_message =
+        "School seeding: done processing #{filename}.\n"\
+        "#{new_schools.length} new schools#{future_tense_dry_run} added.\n"\
+        "#{updated_schools} schools#{future_tense_dry_run} updated.\n"\
+        "#{errored_schools.length} schools#{future_tense_dry_run} updated failed.\n"\
+        "#{unchanged_schools} schools#{future_tense_dry_run} unchanged (school considered unchanged if only update was adding new columns included in this import).\n"\
+        "#{duplicate_schools.length} duplicate schools#{future_tense_dry_run} skipped.\n"\
+        "State CS offerings#{future_tense_dry_run} deleted: #{state_cs_offerings_deleted_count}, state CS offerings#{future_tense_dry_run} reloaded: #{state_cs_offerings_reloaded_count}\n"
+
+      if updated_schools_attribute_frequency.any?
         summary_message <<
-          "Schools#{future_tense_dry_run} added:\n"\
-          "#{new_schools.map {|school| school[:name] + ' ' + school[:id].to_s}.join("\n")}\n"
+          "Among updated schools, these attributes #{is_dry_run ? 'will be' : 'were'} updated:\n"\
+          "#{updated_schools_attribute_frequency.sort_by {|_, v| v}.
+            reverse.
+            map {|attribute, frequency| attribute + ': ' + frequency.to_s}.join("\n")}\n"
       end
 
-      if duplicate_schools.any?
-        summary_message <<
-          "Duplicate schools#{future_tense_dry_run} skipped:\n"\
-          "#{duplicate_schools.map {|school| school[:name] + ' ' + school[:id]}.join("\n")}"
+      # More verbose logging in dry run
+      if is_dry_run
+        if new_schools.any?
+          summary_message <<
+            "Schools#{future_tense_dry_run} added:\n"\
+            "#{new_schools.map {|school| school[:name] + ' ' + school[:id].to_s}.join("\n")}\n"
+        end
+
+        if errored_schools.any?
+          summary_message <<
+            "These schools failed to update:\n"\
+            "#{errored_schools.map {|school| school[:name] + ' ' + school[:id].to_s}.join("\n")}\n"
+        end
+
+        if duplicate_schools.any?
+          summary_message <<
+            "Duplicate schools#{future_tense_dry_run} skipped:\n"\
+            "#{duplicate_schools.map {|school| school[:name] + ' ' + school[:id]}.join("\n")}"
+        end
       end
-    end
 
-    CDO.log.info summary_message
+      CDO.log.info summary_message
+
+      # Raise an error so that the db transaction rolls back
+      raise "This was a dry run" if is_dry_run
```

Comment:
Ooh I like this, so dry runs are actually interacting with the database. I think we'd want more specific rescue clause then, right?

Reply:
I don't think the rescues above would catch this error due to scope, but ya, made the above more specific

Reply:
Oh I didn't mean rescuing this error, I meant that I think we should update the above one, which it sounds like you are doing 👍